clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -120,12 +116,18 @@ public void subscribe(Collection<String> topics, ConsumerRebalanceListener liste
         changeSubscription(topics);
     }
 
-    public void changeSubscription(Collection<String> topicsToSubscribe) {
-        if (!this.subscription.equals(new HashSet<>(topicsToSubscribe))) {
-            this.subscription.clear();
-            this.subscription.addAll(topicsToSubscribe);
+    public void subscribeFromPattern(Set<String> topics) {
+        if (subscriptionType != SubscriptionType.AUTO_PATTERN)
+            throw new IllegalArgumentException("Attempt to subscribe from pattern while subscription type set to " +
+                    subscriptionType);
```

Comment:
Actually the import check is here: https://github.com/hachikuji/kafka/blob/7971c2f45fc399ec8c5d2b628fdd6c575b1b2745/clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java#L160. There shouldn't be any reason for this exception to be thrown, supposing we don't break the code in the future.